zadania/Program.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System.Collections.Generic;
+
+int width = 20;
+int height = 20;
+int n = 15000;
+double[,] generateTableDensity(int width, int height)
+{
+    double[,] densityTable = new double[width, height];
+    for (int x = 0; x < width; x++)
+    {
+        for (int y = 0; y < height; y++)
+        {
+            densityTable[x, y] = ((x - 5) * (x - 5) + (y - 5) * (y - 5)) * (((x - 1) * (x - 1) + (y - 3) * (y - 3)) / (8.0)) + 4;
+        }
+    }
+    return densityTable;
+}
+double[,] generateTableDesirability(int width, int height)
+{
+    double[,] desirabilityTable = new double[width, height];
+    for (int x = 0; x < width; x++)
+    {
+        for (int y = 0; y < height; y++)
+        {
+            desirabilityTable[x, y] = -Math.Abs((x - width / 2.0) * (y - height / 2.0)) + (width / 2.0 * height / 2.0);
+        }
+    }
+    return desirabilityTable;
+}
+List<Tuple<double,int,int>> generateHubsDensity(double[,] densityTable, int masterRandomNumber, Random random)
+{
+    List<Tuple<double, int, int>> hubs = new List<Tuple<double, int, int>>();
+    for(int x = 0; x<masterRandomNumber; x++)
+    {
+        int h = random.Next(height-1);
+        int w = random.Next(width-1);
+        hubs.Add( new Tuple<double, int, int>(densityTable[h, w], h, w));
```

Comment:
Wylosowane huby mogą się powtórzyć. Zabezpiecz się przed tym sprawdzający czy w liście hubs jest już dodany taki hub. Zrobisz to metodą `Contains`